File: back/src/main/java/com/back/domain/ticket/dto/TicketVerifyResponse.java
package com.back.domain.ticket.dto;

import com.back.domain.ticket.entity.Ticket;

import java.time.LocalDateTime;

public record TicketVerifyResponse(
        String concertName,
        String venueName,
        LocalDateTime scheduleDate,
        String seatNumber,
        boolean isValid
) {
    public static TicketVerifyResponse from(Ticket ticket) {
        return new TicketVerifyResponse(
                ticket.getSchedule().getConcert().getConcertName(),
                ticket.getSchedule().getVenue().getVenueName(),
                ticket.getSchedule().getScheduleDate(),
                ticket.getScheduleSeat().getSeatNumber(),
                ticket.isValid()
        );
    }
}

File: back/src/main/kotlin/com/back/domain/ticket/dto/TicketVerifyResponse.kt
package com.back.domain.ticket.dto

import com.back.domain.ticket.entity.Ticket
import java.time.LocalDateTime

data class TicketVerifyResponse(
    val concertName: String,
    val venueName: String,
    val scheduleDate: LocalDateTime,
    val seatNumber: String,
    val isValid: Boolean
) {
    companion object {
        fun from(ticket: Ticket): TicketVerifyResponse {
            return TicketVerifyResponse(
                concertName = ticket.schedule.concert.concertName,
                venueName = ticket.schedule.venue.venueName,
                scheduleDate = ticket.schedule.scheduleDate,
                seatNumber = ticket.scheduleSeat.seatNumber,
                isValid = ticket.isValid
            )
        }
    }
}
